controllers/volumereplication_controller.go

```diff
@@ -317,6 +328,50 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+
+	logger := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForVolumeReplicationResource(logger, volumeReplicationClass)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplicationClass")
+		return err
+	}
+
+	err = r.waitForVolumeReplicationResource(logger, volumeReplication)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplication")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationResource(logger logr.Logger, resourceName string) error {
+
+	isVolumeReplicationResourceExists := false
+	unstructuredResource := &unstructured.UnstructuredList{}
+	unstructuredResource.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   replicationv1alpha1.GroupVersion.Group,
+		Kind:    resourceName,
+		Version: replicationv1alpha1.GroupVersion.Version,
+	})
+	for ! isVolumeReplicationResourceExists {
+		err := r.Client.List(context.TODO(), unstructuredResource)
+		if err != nil {
+			if meta.IsNoMatchError(err){
+				logger.Info("Resource", resourceName, "does not exist")
+				time.Sleep(5 * time.Second)
+			} else {
+				logger.Error(err, "Got an unexpected error while waiting for", "Resource", resourceName)
```

Comment:
```suggestion
				logger.Error(err, "got an unexpected error while waiting for resource", "Resource", resourceName)
```

lets start with small case for consistency with log messages.

Reply:
done